File: go/types/msg.go
package types 

import "encoding/json"


func UnmarshalMsg(data []byte) (Msg, error) {
	var r Msg
	err := json.Unmarshal(data, &r)
	return r, err
}

func (r *Msg) Marshal() ([]byte, error) {
	return json.Marshal(r)
}


/*
炸群

{
  "result": false,
  "request": "/groupchat/query_messages.json",
  "error_code": 21202,
  "error": "不允许此项操作，有疑问请咨询 @粉丝群"
}
*/

type MsgError struct {
	Result    bool   `json:"result"`    
	Request   string `json:"request"`   
	ErrorCode int64  `json:"error_code"`
	Error     string `json:"error"`     
}


type Msg struct {
	LastReadMid     int64         `json:"last_read_mid"`   
	SignificantMsgs []interface{} `json:"significant_msgs"`
	GroupTips       []GroupTip    `json:"group_tips"`      
	Messages        []Messages    `json:"messages"`        
	Result          bool          `json:"result"`          
	Ts              int64         `json:"ts"`              
}

type GroupTip struct {
	GroupID         int64  `json:"group_id"`        
	ContentTemplate string `json:"content_template"`
	Time            int64  `json:"time"`            
	Type            int64  `json:"type"`            
	TipsID          int64  `json:"tips_id"`         
	ContentData     string `json:"content_data"`    
	Content         string `json:"content"`         
}

type Messages struct {
	Annotations  Annotations `json:"annotations"`           
	Appid        int64       `json:"appid"`                 
	Content      string      `json:"content"`               
	FaithStatus  *int64      `json:"faith_status,omitempty"`
	Fids         []int64     `json:"fids"`                  
	FromUid      int64       `json:"from_uid"`              
	FromUser     FromUser    `json:"from_user"`             
	Gid          int64       `json:"gid"`                   
	ID           int64       `json:"id"`                    
	Mblogid      string      `json:"mblogid"`               
	MediaType    int64       `json:"media_type"`            
	RecallStatus int64       `json:"recall_status"`         
	Time         int64       `json:"time"`                  
	Type         int64       `json:"type"`                  
	URLObjects   []URLObject `json:"url_objects"`           
}

type Annotations struct {
	Gdid         string `json:"gdid"`         
	IncludeVideo int64  `json:"include_video"`
	SendFrom     string `json:"send_from"`    
	Spr          string `json:"spr"`          
}

type FromUser struct {
	AvatarLarge     string `json:"avatar_large"`     
	FollowersCount  int64  `json:"followers_count"`  
	FriendsCount    int64  `json:"friends_count"`    
	ID              int64  `json:"id"`               
	Level           int64  `json:"level"`            
	ProfileImageURL string `json:"profile_image_url"`
	ProfileURL      string `json:"profile_url"`      
	ScreenName      string `json:"screen_name"`      
	Verified        bool   `json:"verified"`         
}

type URLObject struct {
	AssoLikeCount         int64  `json:"asso_like_count"`         
	CardInfoUnIntegrity   bool   `json:"card_info_un_integrity"`  
	FollowerCount         int64  `json:"follower_count"`          
	Info                  Info   `json:"info"`                    
	IsActionType          bool   `json:"isActionType"`            
	LikeCount             int64  `json:"like_count"`              
	ObjectID              string `json:"object_id"`               
	PageID                string `json:"page_id"`                 
	SearchTopicCount      int64  `json:"search_topic_count"`      
	SearchTopicReadCount  int64  `json:"search_topic_read_count"` 
	Status                Status `json:"status"`                  
	SuperTopicPhotoCount  int64  `json:"super_topic_photo_count"` 
	SuperTopicStatusCount int64  `json:"super_topic_status_count"`
	URLOri                string `json:"url_ori"`                 
}

type Info struct {
	Description  string `json:"description"`  
	EXTStatus    int64  `json:"ext_status"`   
	LastModified int64  `json:"last_modified"`
	Result       bool   `json:"result"`       
	Title        string `json:"title"`        
	Transcode    int64  `json:"transcode"`    
	Type         int64  `json:"type"`         
	URLLong      string `json:"url_long"`     
	URLShort     string `json:"url_short"`    
}

type Status struct {
	Annotations              []Annotation          `json:"annotations"`               
	Appid                    int64                 `json:"appid"`                     
	AttitudesCount           int64                 `json:"attitudes_count"`           
	BizFeature               int64                 `json:"biz_feature"`               
	BizIDS                   []int64               `json:"biz_ids"`                   
	CanEdit                  bool                  `json:"can_edit"`                  
	Cardid                   string                `json:"cardid"`                    
	CommentManageInfo        CommentManageInfo     `json:"comment_manage_info"`       
	CommentsCount            int64                 `json:"comments_count"`            
	ContentAuth              int64                 `json:"content_auth"`              
	CreatedAt                string                `json:"created_at"`                
	DarwinTags               []DarwinTag           `json:"darwin_tags"`               
	ExternSafe               int64                 `json:"extern_safe"`               
	Favorited                bool                  `json:"favorited"`                 
	Geo                      Geo                   `json:"geo"`                       
	GIFIDS                   string                `json:"gif_ids"`                   
	HasActionTypeCard        int64                 `json:"hasActionTypeCard"`         
	HideFlag                 int64                 `json:"hide_flag"`                 
	HotWeiboTags             []interface{}         `json:"hot_weibo_tags"`            
	ID                       int64                 `json:"id"`                        
	Idstr                    string                `json:"idstr"`                     
	InReplyToScreenName      string                `json:"in_reply_to_screen_name"`   
	InReplyToStatusID        string                `json:"in_reply_to_status_id"`     
	InReplyToUserID          string                `json:"in_reply_to_user_id"`       
	IsLongText               bool                  `json:"isLongText"`                
	IsPaid                   bool                  `json:"is_paid"`                   
	IsShowBulletin           int64                 `json:"is_show_bulletin"`          
	MblogVipType             int64                 `json:"mblog_vip_type"`            
	Mblogtype                int64                 `json:"mblogtype"`                 
	Mid                      string                `json:"mid"`                       
	Mlevel                   int64                 `json:"mlevel"`                    
	MoreInfoType             int64                 `json:"more_info_type"`            
	NumberDisplayStrategy    NumberDisplayStrategy `json:"number_display_strategy"`   
	PendingApprovalCount     int64                 `json:"pending_approval_count"`    
	PicIDS                   []string              `json:"pic_ids"`                   
	PicNum                   int64                 `json:"pic_num"`                   
	PicTypes                 string                `json:"pic_types"`                 
	PositiveRecomFlag        int64                 `json:"positive_recom_flag"`       
	RepostsCount             int64                 `json:"reposts_count"`             
	RewardExhibitionType     int64                 `json:"reward_exhibition_type"`    
	Rid                      string                `json:"rid"`                       
	ShowAdditionalIndication int64                 `json:"show_additional_indication"`
	Source                   string                `json:"source"`                    
	SourceAllowclick         int64                 `json:"source_allowclick"`         
	SourceType               int64                 `json:"source_type"`               
	Text                     string                `json:"text"`                      
	TextLength               int64                 `json:"textLength"`                
	TextTagTips              []interface{}         `json:"text_tag_tips"`             
	Truncated                bool                  `json:"truncated"`                 
	User                     User                  `json:"user"`                      
	UserType                 int64                 `json:"userType"`                  
	Visible                  Visible               `json:"visible"`                   
}

type Annotation struct {
	OID  string `json:"oid"` 
	Type string `json:"type"`
}

type CommentManageInfo struct {
	ApprovalCommentType   int64 `json:"approval_comment_type"`  
	CommentPermissionType int64 `json:"comment_permission_type"`
}

type DarwinTag struct {
	BdObjectType  string      `json:"bd_object_type"`
	DisplayName   string      `json:"display_name"`  
	EnterpriseUid interface{} `json:"enterprise_uid"`
	ObjectID      string      `json:"object_id"`     
	ObjectType    string      `json:"object_type"`   
}

type Geo struct {
	Coordinates []float64 `json:"coordinates"`
	Type        string    `json:"type"`       
}

type NumberDisplayStrategy struct {
	ApplyScenarioFlag    int64  `json:"apply_scenario_flag"`    
	DisplayText          string `json:"display_text"`           
	DisplayTextMinNumber int64  `json:"display_text_min_number"`
}

type User struct {
	AllowAllActMsg    bool             `json:"allow_all_act_msg"`  
	AllowAllComment   bool             `json:"allow_all_comment"`  
	AvatarHD          string           `json:"avatar_hd"`          
	AvatarLarge       string           `json:"avatar_large"`       
	Badge             map[string]int64 `json:"badge"`              
	BadgeTop          string           `json:"badge_top"`          
	BIFollowersCount  int64            `json:"bi_followers_count"` 
	BlockApp          int64            `json:"block_app"`          
	BlockWord         int64            `json:"block_word"`         
	Cardid            string           `json:"cardid"`             
	City              string           `json:"city"`               
	Class             int64            `json:"class"`              
	CoverImagePhone   string           `json:"cover_image_phone"`  
	CreatedAt         string           `json:"created_at"`         
	CreditScore       int64            `json:"credit_score"`       
	Description       string           `json:"description"`        
	Domain            string           `json:"domain"`             
	Extend            Extend           `json:"extend"`             
	FavouritesCount   int64            `json:"favourites_count"`   
	FollowMe          bool             `json:"follow_me"`          
	FollowersCount    int64            `json:"followers_count"`    
	Following         bool             `json:"following"`          
	FriendsCount      int64            `json:"friends_count"`      
	Gender            string           `json:"gender"`             
	GeoEnabled        bool             `json:"geo_enabled"`        
	HasAbilityTag     int64            `json:"has_ability_tag"`    
	ID                int64            `json:"id"`                 
	Idstr             string           `json:"idstr"`              
	Insecurity        Insecurity       `json:"insecurity"`         
	IsGuardian        int64            `json:"is_guardian"`        
	IsTeenager        int64            `json:"is_teenager"`        
	IsTeenagerList    int64            `json:"is_teenager_list"`   
	Lang              string           `json:"lang"`               
	Level             int64            `json:"level"`              
	Like              bool             `json:"like"`               
	LikeMe            bool             `json:"like_me"`            
	Location          string           `json:"location"`           
	Mbrank            int64            `json:"mbrank"`             
	Mbtype            int64            `json:"mbtype"`             
	Name              string           `json:"name"`               
	OnlineStatus      int64            `json:"online_status"`      
	PagefriendsCount  int64            `json:"pagefriends_count"`  
	ProfileImageURL   string           `json:"profile_image_url"`  
	ProfileURL        string           `json:"profile_url"`        
	Province          string           `json:"province"`           
	Ptype             int64            `json:"ptype"`              
	Remark            string           `json:"remark"`             
	ScreenName        string           `json:"screen_name"`        
	SpecialFollow     bool             `json:"special_follow"`     
	Star              int64            `json:"star"`               
	StatusesCount     int64            `json:"statuses_count"`     
	StoryReadState    int64            `json:"story_read_state"`   
	TabManage         string           `json:"tab_manage"`         
	Type              int64            `json:"type"`               
	Ulevel            int64            `json:"ulevel"`             
	Urank             int64            `json:"urank"`              
	URL               string           `json:"url"`                
	UserAbility       int64            `json:"user_ability"`       
	UserLimit         int64            `json:"user_limit"`         
	VclubMember       int64            `json:"vclub_member"`       
	Verified          bool             `json:"verified"`           
	VerifiedReason    string           `json:"verified_reason"`    
	VerifiedReasonURL string           `json:"verified_reason_url"`
	VerifiedSource    string           `json:"verified_source"`    
	VerifiedSourceURL string           `json:"verified_source_url"`
	VerifiedTrade     string           `json:"verified_trade"`     
	VerifiedType      int64            `json:"verified_type"`      
	VideoStatusCount  int64            `json:"video_status_count"` 
	Weihao            string           `json:"weihao"`             
}

type Extend struct {
	Mbprivilege string  `json:"mbprivilege"`
	Privacy     Privacy `json:"privacy"`    
}

type Privacy struct {
	Mobile int64 `json:"mobile"`
}

type Insecurity struct {
	SexualContent bool `json:"sexual_content"`
}

type Visible struct {
	ListID int64 `json:"list_id"`
	Type   int64 `json:"type"`   
}

File: server/src/msg.rs

// Example code that deserializes and serializes the model.
// extern crate serde;
// #[macro_use]
// extern crate serde_derive;
// extern crate serde_json;
//
// use generated_module::[object Object];
//
// fn main() {
//     let json = r#"{"answer": 42}"#;
//     let model: [object Object] = serde_json::from_str(&json).unwrap();
// }

extern crate serde_json;
use std::collections::HashMap;



#[derive(Serialize, Deserialize)]
pub struct Msg{
    #[serde(rename = "last_read_mid")]
    last_read_mid: i64,

    #[serde(rename = "significant_msgs")]
    significant_msgs: Vec<Option<serde_json::Value>>,

    #[serde(rename = "group_tips")]
    group_tips: Vec<Option<serde_json::Value>>,

    #[serde(rename = "messages")]
    messages: Vec<Message>,

    #[serde(rename = "result")]
    result: bool,

    #[serde(rename = "ts")]
    ts: i64,
}


#[derive(Serialize, Deserialize)]
pub struct Message{
    #[serde(rename = "annotations")]
    annotations: Annotations,

    #[serde(rename = "appid")]
    appid: i64,

    #[serde(rename = "content")]
    content: String,

    #[serde(rename = "faith_status")]
    faith_status: Option<i64>,

    #[serde(rename = "fids")]
    fids: Option<Vec<i64>>,

    #[serde(rename = "from_uid")]
    from_uid: i64,

    #[serde(rename = "from_user")]
    from_user: FromUser,

    #[serde(rename = "gid")]
    gid: i64,

    #[serde(rename = "id")]
    id: i64,

    #[serde(rename = "mblogid")]
    mblogid: String,

    #[serde(rename = "media_type")]
    media_type: i64,

    #[serde(rename = "recall_status")]
    recall_status: i64,

    #[serde(rename = "time")]
    time: i64,

    #[serde(rename = "type")]
    top_level_type: i64,

    #[serde(rename = "url_objects")]
    url_objects: Option<Vec<UrlObject>>,
}

#[derive(Serialize, Deserialize)]
pub struct Annotations {
    #[serde(rename = "gdid")]
    gdid: String,

    #[serde(rename = "include_video")]
    include_video: i64,

    #[serde(rename = "send_from")]
    send_from: String,

    #[serde(rename = "spr")]
    spr: String,
}

#[derive(Serialize, Deserialize)]
pub struct FromUser {
    #[serde(rename = "avatar_large")]
    avatar_large: String,

    #[serde(rename = "followers_count")]
    followers_count: i64,

    #[serde(rename = "friends_count")]
    friends_count: i64,

    #[serde(rename = "id")]
    id: i64,

    #[serde(rename = "level")]
    level: i64,

    #[serde(rename = "profile_image_url")]
    profile_image_url: String,

    #[serde(rename = "profile_url")]
    profile_url: String,

    #[serde(rename = "screen_name")]
    screen_name: String,

    #[serde(rename = "verified")]
    verified: bool,
}

#[derive(Serialize, Deserialize)]
pub struct UrlObject {
    #[serde(rename = "asso_like_count")]
    asso_like_count: i64,

    #[serde(rename = "card_info_un_integrity")]
    card_info_un_integrity: bool,

    #[serde(rename = "follower_count")]
    follower_count: i64,

    #[serde(rename = "info")]
    info: Info,

    #[serde(rename = "isActionType")]
    is_action_type: bool,

    #[serde(rename = "like_count")]
    like_count: i64,

    #[serde(rename = "object_id")]
    object_id: String,

    #[serde(rename = "page_id")]
    page_id: String,

    #[serde(rename = "search_topic_count")]
    search_topic_count: i64,

    #[serde(rename = "search_topic_read_count")]
    search_topic_read_count: i64,

    #[serde(rename = "status")]
    status: Status,

    #[serde(rename = "super_topic_photo_count")]
    super_topic_photo_count: i64,

    #[serde(rename = "super_topic_status_count")]
    super_topic_status_count: i64,

    #[serde(rename = "url_ori")]
    url_ori: String,
}

#[derive(Serialize, Deserialize)]
pub struct Info {
    #[serde(rename = "description")]
    description: String,

    #[serde(rename = "ext_status")]
    ext_status: i64,

    #[serde(rename = "last_modified")]
    last_modified: i64,

    #[serde(rename = "result")]
    result: bool,

    #[serde(rename = "title")]
    title: String,

    #[serde(rename = "transcode")]
    transcode: i64,

    #[serde(rename = "type")]
    info_type: i64,

    #[serde(rename = "url_long")]
    url_long: String,

    #[serde(rename = "url_short")]
    url_short: String,
}

#[derive(Serialize, Deserialize)]
pub struct Status {
    #[serde(rename = "annotations")]
    annotations: Vec<Annotation>,

    #[serde(rename = "appid")]
    appid: i64,

    #[serde(rename = "attitudes_count")]
    attitudes_count: i64,

    #[serde(rename = "biz_feature")]
    biz_feature: i64,

    #[serde(rename = "biz_ids")]
    biz_ids: Vec<i64>,

    #[serde(rename = "can_edit")]
    can_edit: bool,

    #[serde(rename = "cardid")]
    cardid: String,

    #[serde(rename = "comment_manage_info")]
    comment_manage_info: CommentManageInfo,

    #[serde(rename = "comments_count")]
    comments_count: i64,

    #[serde(rename = "content_auth")]
    content_auth: i64,

    #[serde(rename = "created_at")]
    created_at: String,

    #[serde(rename = "darwin_tags")]
    darwin_tags: Vec<DarwinTag>,

    #[serde(rename = "extern_safe")]
    extern_safe: i64,

    #[serde(rename = "favorited")]
    favorited: bool,

    #[serde(rename = "geo")]
    geo: Geo,

    #[serde(rename = "gif_ids")]
    gif_ids: String,

    #[serde(rename = "hasActionTypeCard")]
    has_action_type_card: i64,

    #[serde(rename = "hide_flag")]
    hide_flag: i64,

    #[serde(rename = "hot_weibo_tags")]
    hot_weibo_tags: Vec<Option<serde_json::Value>>,

    #[serde(rename = "id")]
    id: i64,

    #[serde(rename = "idstr")]
    idstr: String,

    #[serde(rename = "in_reply_to_screen_name")]
    in_reply_to_screen_name: String,

    #[serde(rename = "in_reply_to_status_id")]
    in_reply_to_status_id: String,

    #[serde(rename = "in_reply_to_user_id")]
    in_reply_to_user_id: String,

    #[serde(rename = "isLongText")]
    is_long_text: bool,

    #[serde(rename = "is_paid")]
    is_paid: bool,

    #[serde(rename = "is_show_bulletin")]
    is_show_bulletin: i64,

    #[serde(rename = "mblog_vip_type")]
    mblog_vip_type: i64,

    #[serde(rename = "mblogtype")]
    mblogtype: i64,

    #[serde(rename = "mid")]
    mid: String,

    #[serde(rename = "mlevel")]
    mlevel: i64,

    #[serde(rename = "more_info_type")]
    more_info_type: i64,

    #[serde(rename = "number_display_strategy")]
    number_display_strategy: NumberDisplayStrategy,

    #[serde(rename = "pending_approval_count")]
    pending_approval_count: i64,

    #[serde(rename = "pic_ids")]
    pic_ids: Vec<String>,

    #[serde(rename = "pic_num")]
    pic_num: i64,

    #[serde(rename = "pic_types")]
    pic_types: String,

    #[serde(rename = "positive_recom_flag")]
    positive_recom_flag: i64,

    #[serde(rename = "reposts_count")]
    reposts_count: i64,

    #[serde(rename = "reward_exhibition_type")]
    reward_exhibition_type: i64,

    #[serde(rename = "rid")]
    rid: String,

    #[serde(rename = "show_additional_indication")]
    show_additional_indication: i64,

    #[serde(rename = "source")]
    source: String,

    #[serde(rename = "source_allowclick")]
    source_allowclick: i64,

    #[serde(rename = "source_type")]
    source_type: i64,

    #[serde(rename = "text")]
    text: String,

    #[serde(rename = "textLength")]
    text_length: i64,

    #[serde(rename = "text_tag_tips")]
    text_tag_tips: Vec<Option<serde_json::Value>>,

    #[serde(rename = "truncated")]
    truncated: bool,

    #[serde(rename = "user")]
    user: User,

    #[serde(rename = "userType")]
    user_type: i64,

    #[serde(rename = "visible")]
    visible: Visible,
}

#[derive(Serialize, Deserialize)]
pub struct Annotation {
    #[serde(rename = "oid")]
    oid: String,

    #[serde(rename = "type")]
    annotation_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct CommentManageInfo {
    #[serde(rename = "approval_comment_type")]
    approval_comment_type: i64,

    #[serde(rename = "comment_permission_type")]
    comment_permission_type: i64,
}

#[derive(Serialize, Deserialize)]
pub struct DarwinTag {
    #[serde(rename = "bd_object_type")]
    bd_object_type: String,

    #[serde(rename = "display_name")]
    display_name: String,

    #[serde(rename = "enterprise_uid")]
    enterprise_uid: Option<serde_json::Value>,

    #[serde(rename = "object_id")]
    object_id: String,

    #[serde(rename = "object_type")]
    object_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct Geo {
    #[serde(rename = "coordinates")]
    coordinates: Vec<f64>,

    #[serde(rename = "type")]
    geo_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct NumberDisplayStrategy {
    #[serde(rename = "apply_scenario_flag")]
    apply_scenario_flag: i64,

    #[serde(rename = "display_text")]
    display_text: String,

    #[serde(rename = "display_text_min_number")]
    display_text_min_number: i64,
}

#[derive(Serialize, Deserialize)]
pub struct User {
    #[serde(rename = "allow_all_act_msg")]
    allow_all_act_msg: bool,

    #[serde(rename = "allow_all_comment")]
    allow_all_comment: bool,

    #[serde(rename = "avatar_hd")]
    avatar_hd: String,

    #[serde(rename = "avatar_large")]
    avatar_large: String,

    #[serde(rename = "badge")]
    badge: HashMap<String, i64>,

    #[serde(rename = "badge_top")]
    badge_top: String,

    #[serde(rename = "bi_followers_count")]
    bi_followers_count: i64,

    #[serde(rename = "block_app")]
    block_app: i64,

    #[serde(rename = "block_word")]
    block_word: i64,

    #[serde(rename = "cardid")]
    cardid: String,

    #[serde(rename = "city")]
    city: String,

    #[serde(rename = "class")]
    class: i64,

    #[serde(rename = "cover_image_phone")]
    cover_image_phone: String,

    #[serde(rename = "created_at")]
    created_at: String,

    #[serde(rename = "credit_score")]
    credit_score: i64,

    #[serde(rename = "description")]
    description: String,

    #[serde(rename = "domain")]
    domain: String,

    #[serde(rename = "extend")]
    extend: Extend,

    #[serde(rename = "favourites_count")]
    favourites_count: i64,

    #[serde(rename = "follow_me")]
    follow_me: bool,

    #[serde(rename = "followers_count")]
    followers_count: i64,

    #[serde(rename = "following")]
    following: bool,

    #[serde(rename = "friends_count")]
    friends_count: i64,

    #[serde(rename = "gender")]
    gender: String,

    #[serde(rename = "geo_enabled")]
    geo_enabled: bool,

    #[serde(rename = "has_ability_tag")]
    has_ability_tag: i64,

    #[serde(rename = "id")]
    id: i64,

    #[serde(rename = "idstr")]
    idstr: String,

    #[serde(rename = "insecurity")]
    insecurity: Insecurity,

    #[serde(rename = "is_guardian")]
    is_guardian: i64,

    #[serde(rename = "is_teenager")]
    is_teenager: i64,

    #[serde(rename = "is_teenager_list")]
    is_teenager_list: i64,

    #[serde(rename = "lang")]
    lang: String,

    #[serde(rename = "level")]
    level: i64,

    #[serde(rename = "like")]
    like: bool,

    #[serde(rename = "like_me")]
    like_me: bool,

    #[serde(rename = "location")]
    location: String,

    #[serde(rename = "mbrank")]
    mbrank: i64,

    #[serde(rename = "mbtype")]
    mbtype: i64,

    #[serde(rename = "name")]
    name: String,

    #[serde(rename = "online_status")]
    online_status: i64,

    #[serde(rename = "pagefriends_count")]
    pagefriends_count: i64,

    #[serde(rename = "profile_image_url")]
    profile_image_url: String,

    #[serde(rename = "profile_url")]
    profile_url: String,

    #[serde(rename = "province")]
    province: String,

    #[serde(rename = "ptype")]
    ptype: i64,

    #[serde(rename = "remark")]
    remark: String,

    #[serde(rename = "screen_name")]
    screen_name: String,

    #[serde(rename = "special_follow")]
    special_follow: bool,

    #[serde(rename = "star")]
    star: i64,

    #[serde(rename = "statuses_count")]
    statuses_count: i64,

    #[serde(rename = "story_read_state")]
    story_read_state: i64,

    #[serde(rename = "tab_manage")]
    tab_manage: String,

    #[serde(rename = "type")]
    user_type: i64,

    #[serde(rename = "ulevel")]
    ulevel: i64,

    #[serde(rename = "urank")]
    urank: i64,

    #[serde(rename = "url")]
    url: String,

    #[serde(rename = "user_ability")]
    user_ability: i64,

    #[serde(rename = "user_limit")]
    user_limit: i64,

    #[serde(rename = "vclub_member")]
    vclub_member: i64,

    #[serde(rename = "verified")]
    verified: bool,

    #[serde(rename = "verified_reason")]
    verified_reason: String,

    #[serde(rename = "verified_reason_url")]
    verified_reason_url: String,

    #[serde(rename = "verified_source")]
    verified_source: String,

    #[serde(rename = "verified_source_url")]
    verified_source_url: String,

    #[serde(rename = "verified_trade")]
    verified_trade: String,

    #[serde(rename = "verified_type")]
    verified_type: i64,

    #[serde(rename = "video_status_count")]
    video_status_count: i64,

    #[serde(rename = "weihao")]
    weihao: String,
}

#[derive(Serialize, Deserialize)]
pub struct Extend {
    #[serde(rename = "mbprivilege")]
    mbprivilege: String,

    #[serde(rename = "privacy")]
    privacy: Privacy,
}

#[derive(Serialize, Deserialize)]
pub struct Privacy {
    #[serde(rename = "mobile")]
    mobile: i64,
}

#[derive(Serialize, Deserialize)]
pub struct Insecurity {
    #[serde(rename = "sexual_content")]
    sexual_content: bool,
}

#[derive(Serialize, Deserialize)]
pub struct Visible {
    #[serde(rename = "list_id")]
    list_id: i64,

    #[serde(rename = "type")]
    visible_type: i64,
}
